add_lcsi function and tests

SaeedR1987 opened the pull request.

SaeedR1987 committed:
Adding add_lcsi

add_lcsi function and tests

SaeedR1987 committed:
Merge branch 'master' of https://github.com/SaeedR1987/addindicators

SaeedR1987 committed:
add rlang to imports

assil-benamor commented on R/add_lcsi.R:
```diff
@@ -1,20 +1,179 @@
+
+df <- .dataset
+
+  #If NULL, set a standard default values for LCSI responses.
```
I am thinking maybe it's better for the default values to use the names from the HQ MSNA template:
yes
no_had_no_need
no_exhausted
not_applicable

assil-benamor commented on R/add_lcsi.R:
```diff
@@ -1,20 +1,179 @@
+  #If NULL, set a standard default values for LCSI responses.
+
+  if(is.null(yes_val)) {yes_val = "1"}
+  if(is.null(yes_val)) {no_val = "2"}
```
I think this should be:
if(is.null(no_val)) {no_val = "2"}
if(is.null(exhausted_val)) {exhausted_val = "3"}
if(is.null(not_applicable_val)) {not_applicable_val = "4"}
[You used yes_val for all 4 checks]

assil-benamor commented on R/add_lcsi.R:
```diff
@@ -1,20 +1,179 @@
+
+  # Warning 1: If < 4 unique responses observed, and
+
+  if(length(lcs_codes) < 4 & !all(a,b,c,d)) {
```
Can you also check that the lcs codes are different? 
For example I was able to call the function with these params and it went through 
`add_lcsi(.dataset = input_data1,
         lcsi_stress_vars = c("stress1", "stress2", "stress3", "stress4"),
         lcsi_crisis_vars = c("crisis1", "crisis2", "crisis3"),
         lcsi_emergency_vars = c("emergency1", "emergency2", "emergency3"),
         yes_val = "Yes",
         no_val = "No",
         exhausted_val = "Exhausted",
         not_applicable_val = "Exhausted")`

SaeedR1987 committed:
Update add_lcsi

Updated checking for duplicate parameter inputs, and using standard response values in line with HQ MSNA indicator bank.
